### PR TITLE
Script to amend monthly rollup microdata

### PR DESCRIPTION
**Pull request by nmdefries**
### Description
Add state field based on county FIPS. Change name of `wave` field to
`version`.

### Changelog
- Add script `microdata_add_state_col__rename_wave.R`

### CODE REVIEW

**Review by capnrefsmmat — COMMENTED** [review]
I think this will be fine, just slightly more complicated than it has to be, since the FIPS conversion can be easier.

Not sure if it's worth changing the conversion if it already works. Could you maybe add an assertion after line 42 that checks that all rows with FIPS codes get a non-NA state? That should definitely happen, but if there's something weird about our mapping files (like with territories), we could get issues, and we don't want to have missingness because of that.

File: facebook/microdata_add_state_col__rename_wave.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    # stop readr from thinking commas = thousand separators,
+    # and from inferring column types incorrectly
+    message("reading data in")
+    data <- read_csv(file.path(input_dir, fname), locale = locale(grouping_mark = ""),
+             col_types = cols(
+               .default = col_character())) %>%
+      rename(version = .data$wave)
+
+    # Add state column based on county FIPS code.
+    data <- left_join(data, state_county_map, by="fips")
```
[the remark this answers is not in the document]
Comment (nmdefries) [reply facebook/microdata_add_state_col__rename_wave.R:42]: :facepalm: and I thought this *was* fairly straightforward

File: facebook/amend_monthly_microdata.R [hunk 2]
```diff
@@ -47,12 +53,43 @@ amend_microdata <- function(input_dir, output_dir, static_dir, pattern = ".*[.]c
...
+
+  # clean the ZIP data
+  data$zip5 <- stri_replace_all(data$zip5, "", regex = " *")
+  data$zip5 <- stri_replace(data$zip5, "", regex ="-.*")
+
+  # some people enter 9-digit ZIPs, which could make them easily identifiable in
+  # the individual output files. rather than truncating to 5 digits -- which may
+  # turn nonsense entered by some respondents into a valid ZIP5 -- we simply
+  # replace these ZIPs with NA.
+  data$zip5 <- ifelse(nchar(data$zip5) > 5, NA_character_,
```
Comment (nmdefries) [inline facebook/amend_monthly_microdata.R:77]: The logic for these two functions is borrowed from `delphiFacebook/R/responses.R`, but I'm wondering if this line is supposed to be handling A3. We drop `zip5` anyway, so nulling it out here seems unnecessary. @capnrefsmmat

Comment (capnrefsmmat) [reply facebook/amend_monthly_microdata.R:77]: `filter_complete_responses` does the other half of the work: 

https://github.com/cmu-delphi/covidcast-indicators/blob/3590e64ea10aade257030b6f8f0815f9ccc3f30b/facebook/delphiFacebook/R/responses.R#L772-L777

We join with the ZIP metadata based on zip5, then blank out A3 based on the results (not zip5). Don't ask me why we need to use two different columns; undoubtedly the code did that in early 2020 and the logic just got ported over to this version

**Review by capnrefsmmat — APPROVED** [review]

**Pull request merged**